kafka/common.go

```diff
@@ -217,19 +218,20 @@ func (cfg *CommonConfig) finalize() error {
 		cfg.TLS = &tls.Config{}
 		tlsInsecure := os.Getenv("KAFKA_TLS_INSECURE") == "true"
 		caCertPath := os.Getenv("KAFKA_TLS_CA_CERT_PATH")
-		if tlsInsecure && caCertPath != "" {
+		if tlsInsecure && (caCertPath != "" || certPath != "" || keyPath != "") {
 			errs = append(errs, errors.New(
-				"kafka: cannot set both KAFKA_TLS_INSECURE and KAFKA_TLS_CA_CERT_PATH",
+				"kafka: cannot set both KAFKA_TLS_INSECURE and KAFKA_TLS_CA_CERT_PATH, KAFKA_TLS_CERT_PATH, or KAFKA_TLS_KEY_PATH",
 			))
 			break
 		}
 		if tlsInsecure {
 			cfg.TLS.InsecureSkipVerify = true
 		}
 		if caCertPath != "" {
-			// Auto-configure a dialer that reloads the CA cert when the file
-			// changes.
-			dialFn, err := newCertReloadingDialer(caCertPath, cfg.TLS)
+			// Set a dialer that reloads the CA cert when the file changes.
+			dialFn, err := newCertReloadingDialer(
+				caCertPath, certPath, keyPath, 30*time.Second, cfg.TLS,
```

Comment:
Question: what happens in between the time that the certificate changed and it is reloaded?
Do we risk to lose data because the connection is rejected?